server.go

```diff
@@ -2831,23 +2844,46 @@ func (s *server) fetchNodeAdvertisedAddr(pub *btcec.PublicKey) (net.Addr, error)
 	return node.Addresses[0], nil
 }
 
-// disableChannel disables a channel, resulting in it not being able to forward
-// payments. This is done by sending a new channel update across the network
-// with the disabled flag set.
-func (s *server) disableChannel(op wire.OutPoint) error {
+// announceChanStatus disables a channel if disabled=true, otherwise activates
+// it. This is done by sending a new channel update across the network with the
+// disabled flag set accordingly. The result of disabling the channel is it not
+// being able to forward payments.
+func (s *server) announceChanStatus(op wire.OutPoint, disabled bool) error {
+	s.sendDisabledMtx.Lock()
+	defer s.sendDisabledMtx.Unlock()
+
+	// If we have already sent out an update reflecting the current status,
+	// skip this channel.
+	alreadyDisabled, ok := s.sentDisabled[op]
+	if ok && alreadyDisabled == disabled {
+		return nil
+	}
+
+	srvrLog.Debugf("Announcing channel(%v) disabled=%v", op, disabled)
+
 	// Retrieve the latest update for this channel. We'll use this
 	// as our starting point to send the new update.
 	chanUpdate, err := s.fetchLastChanUpdateByOutPoint(op)
 	if err != nil {
 		return err
 	}
 
-	// Set the bit responsible for marking a channel as disabled.
-	chanUpdate.Flags |= lnwire.ChanUpdateDisabled
+	if disabled {
+		// Set the bit responsible for marking a channel as disabled.
+		chanUpdate.Flags |= lnwire.ChanUpdateDisabled
+	} else {
+		// Clear the bit responsible for marking a channel as disabled.
+		chanUpdate.Flags &= ^lnwire.ChanUpdateDisabled
+	}
 
 	// We must now update the message's timestamp and generate a new
 	// signature.
-	chanUpdate.Timestamp = uint32(time.Now().Unix())
+	newTimestamp := uint32(time.Now().Unix())
+	if newTimestamp <= chanUpdate.Timestamp {
```

Comment:
Good catch

Reply:
Hmm would like to know how it's possible for the timestamp to be in the future.

Reply:
Not that it's in the future, but if two updates closely follow each other, if we don't bump the timestamp, then it won't propagate. For the `<`, older version of go don't enforce monotonic clocks.